src/pages/docs/guides/ai-transport/openai-sdk.mdx

```diff
@@ -0,0 +1,664 @@
+---
+title: "Guide: Token streaming using Ably and the OpenAI SDK"
+meta_description: "Stream AI model responses from the OpenAI SDK over Ably with real-time token delivery."
+meta_keywords: "AI, token streaming, OpenAI, Responses API, AI transport, Ably, realtime"
+---
+
+This guide demonstrates how to stream token-by-token responses from OpenAI's Responses API over Ably's realtime messaging platform. This pattern enables you to broadcast AI model responses to multiple clients simultaneously, with clear message boundaries for handling response lifecycle.
+
+By using Ably to distribute tokens from the OpenAI SDK, you can scale to thousands of concurrent subscribers, decouple your AI inference from client connections, and maintain reliable message delivery with ordering guarantees. This guide implements the [message-per-token with explicit start/stop events](/docs/ai-transport/features/token-streaming/message-per-token#explicit-events) pattern, which provides clear lifecycle management for each AI response.
+
+<Aside data-type="note">
+This guide focuses on OpenAI's Responses API. For general concepts about token streaming patterns, see the [message-per-token](/docs/ai-transport/features/token-streaming/message-per-token) guide.
+</Aside>
+
+## Prerequisites <a id="prerequisites"/>
+
+To follow this guide, you'll need:
+
+**Software requirements:**
+- Node.js 20 or higher
+
+We'll be using the OpenAPI SDK v4.x.
+
+**Account requirements:**
+- An OpenAI API key
+- An Ably API key
+
+**Useful links:**
+- [OpenAPI developer quickstart](https://platform.openai.com/docs/quickstart)
+- [Ably JavaScript SDK getting started](/docs/getting-started/javascript)
+
+## Setting up your environment <a id="setup"/>
+
+Create a new NPM package. This will contain our publisher and subscriber code:
+
+<Code>
+```shell
+mkdir ably-openai-example && cd ably-openai-example
+npm init -y
+```
+</Code>
+
+Install the required packages using NPM:
+
+<Code>
+```shell
+npm install openai@^4 ably@^2
+```
+</Code>
+
+Export your OpenAPI key to the environment; the OpenAI SDK will automatically read it later:
+
+<Code>
+```shell
+export OPENAI_API_KEY="your_api_key_here"
+```
+</Code>
+
+## Initializing the clients <a id="setup"/>
+
+### Publisher: Initializing the Ably and OpenAI clients <a id="publisher-setup"/>
+
+Initialize both the Ably and OpenAI clients. The Ably Realtime client maintains a persistent connection to the Ably service, allowing you to publish tokens at high message rates with low latency. The OpenAI client provides access to the Responses API.
+
+Create a new file `publisher.mjs` with the following contents:
+
+<Code>
+```javascript
+import Ably from 'ably';
+import OpenAI from 'openai';
+
+// Initialize Ably Realtime client
+const realtime = new Ably.Realtime({ key: '{{API_KEY}}' });
```

Comment:
The docs site is clever and fills this in with a demo API key, which is very convenient but I think means you need to explicitly tell people that this is their Ably API key in the accompanying comment (same for the subscriber snippet)

Reply:
Clearly that's just a me thing, let's stick with consistency